updating/update-using-custom-machine-config-pools.adoc

```diff
@@ -0,0 +1,220 @@
++
+[NOTE]
+====
+In case of a failure i.e. if the MCP with the new version does not work as expected with the applications, the nodes in the pool can be cordoned and drained. So that the application will not run on these nodes and the extra capacity will help to maintain the quality of service of the applications.
+====
++
+. Pause the MCPs created in the previous step.
++
+[NOTE]
+====
+Pausing the MCP will also pause the kube-apiserver-to-kubelet-signer automatic CA certificates rotation.
+You need to unpause these MCPs before the next automatic CA certificate renewal is scheduled.
+See the link:https://access.redhat.com/articles/5651701[article] to findout how much time you have before the next automatic CA certificate renewal happens.
+The update process including updating the whole worker pool need to complete before the CA certification happens.
+====
++
+. Start the update process. The update process will only update the MCPs which are not paused. That includes the master nodes i.e. the control plane.
++
+. Once the control plane update is completed, unpause one MCP. Unpausing the MCP will start the update process for the pool of nodes. You can check the progress of the update in the web console (In Administrator view -> Administration -> Cluster settings ) as well as running `$ oc get machineconfigpools` CLI command.
++
+[NOTE]
+====
+You can change `maxUnavailable` in a MCP to specify the percentage or the number of machines that can be updating at any given time. The default is 1.
+====
++
+. Test if the applications are working as expected on the newly updated MCP.
+
+. Update the remaining MCPs By unpausing one by one till all worker nodes are updated.
+
+. Unpause the MCPs you had paused in the previous step.
+
+== Steps to create a MCP
+
+. Get the list of worker nodes.
++
+[source,terminal]
+----
+$ oc get -l 'node-role.kubernetes.io/master!=' -o 'jsonpath={range .items[*]}{.metadata.name}{"\n"}{end}' nodes
+----
++
+.. Example:
++
+[source,terminal]
+----
+$ oc get -l 'node-role.kubernetes.io/master!=' -o 'jsonpath={range .items[*]}{.metadata.name}{"\n"}{end}' nodes
+ci-ln-pwnll6b-f76d1-s8t9n-worker-a-s75z4
+ci-ln-pwnll6b-f76d1-s8t9n-worker-b-dglj2
+ci-ln-pwnll6b-f76d1-s8t9n-worker-c-lldbm
+----
+. Add the MCP name as label to the worker node.
++
+[source,terminal]
+----
+$ oc label node <node name> node-role.kubernetes.io/<mcp name>=
+----
+.. Example:
++
+[source,terminal]
+----
+$ oc label node ci-ln-gtrwm8t-f76d1-spbl7-worker-a-xk76k node-role.kubernetes.io/mcp1=
+node/ci-ln-gtrwm8t-f76d1-spbl7-worker-a-xk76k labeled
+----
++
+. Create a the machineconfig pool
++
+[source,yaml]
+----
+$ cat mcp_foo.yaml
+apiVersion: machineconfiguration.openshift.io/v1
+kind: MachineConfigPool
+metadata:
+  name: mcp_foo | <1>
+spec:
+  machineConfigSelector:
+    matchExpressions:
+      - {key: machineconfiguration.openshift.io/role, operator: In, values: [worker,mcp_foo]} | <1>
```

Comment:
this need to be fixed, you are creating `mcp1` as nod label but applying something else here. It will be good idea to try on a cluster and then update it.

Reply:
Fixed